### ISSUE TITLE
Exceptions thrown by functions are not detect

### ISSUE THREAD

**Issue by marcospassos** [pepakriz/phpstan-exception-rules#98]
The rule does not detect exceptions thrown by functions:

```php
/**
 * @throws RuntimeException
 */
function foo() : void {
}

try {
    foo();
} catch (RuntimeException $exception) { // RuntimeException is never thrown in the corresponding try block
}
```

**Comment by marcospassos** [comment]
Fai
